### PR TITLE
feat: Add generic error component

### PR DESCRIPTION
**Pull request by Zerdados**
Takes care of ODDSK-78 and ODDSK-51

Preview for faulty endpoint: https://deploy-preview-80--open-data-dashboard.netlify.app/en/resource/7
Preview for empty data: https://deploy-preview-80--open-data-dashboard.netlify.app/en/resource/8
Preview for not found page: https://deploy-preview-80--open-data-dashboard.netlify.app/en/resource/9

### CODE REVIEW

File: src/messages/de.json [hunk 1]
```diff
@@ -3,6 +3,12 @@
     "title": "Open Data Dashboard",
     "description": "Visualisierungen verschiedener Datensätze der Stadt Karlsruhe"
   },
+  "Visualization": {
+    "dataEmptyTitle": "Ressource beinhaltet keine Daten!",
+    "dataEmptyMessage": "Endpunkt \"{source}\" für Ressource \"{name}\" mit id {id} hat keine/leere Daten zurückgegeben. Melden sie dieses Problem dem Administrator dieser Seite.",
```
Comment (danielptv) [inline src/messages/de.json:8]: This is a matter of taste.

In my opinion, the error message output to the user doesn't need to be this verbose. The content you have in the `errorTitle` would be enough. In the subtitle just put the second sentence (`Falls diese Problem weiterhin besteht ...`).

The detailed output (contents of first subtitle sentence) we could log as error to the console together with the actual exception (which is already logged I think).

File: src/app/error.tsx
Comment (danielptv) [inline src/app/error.tsx:1]: Does this work? We are calling the `Error` component which is using internationalization, but this page does not have a locale context. Maybe I am missing something

File: src/components/error-handling/ErrorComponent.tsx
Comment (danielptv) [inline src/components/error-handling/ErrorComponent.tsx:1]: I would prefer if ErrorComponent and NotFound would be one component, as they share much of the functionality and styling. Would make for a more consistent design.

Probably, we dont't need the button for displaying the actual exception, for reasons explained in a comment below. An admin or a curious user can just look it up in the console

**Review by danielptv — COMMENTED** [review]
Looking good. Works well functionality-wise. UI/UX needs a little bit of finetuning

File: src/messages/de.json (hunk 1 above)
Comment (Zerdados) [reply src/messages/de.json:8]: Good point

File: src/app/error.tsx
Comment (Zerdados) [reply src/app/error.tsx:1]: Oh right, great point. It appears this file was unnecessary

**Review by danielptv — APPROVED** [review]
Looks good.

The only thing I have noticed is that the logged error does not appear in the console. No idea why, thought. Can look into in a follow up ticket if we get the time.

**Pull request merged**